test/libdl.jl

```diff
@@ -171,7 +171,7 @@ let dl = C_NULL
     end
 end
 
-if OS_NAME in (:Linux, :FreeBSD)
+if Sys.KERNEL in (:Linux, :FreeBSD)
```

Comment:
Checking with dlsym defeats the purpose of this test since this is all about checking the given symbol exists.